app/static/js/app.js

```diff
@@ -247,6 +245,53 @@ function sendPastedText(pastedText) {
   document.getElementById("app").focus();
 }
 
+function restoreCursor() {
+  if (
+    "settings" in window &&
+    "cursor" in window.settings &&
+    window.settings.cursor != null
+  ) {
+    setCursor(window.settings.cursor);
+  } else {
+    setCursor(3);
+  }
+}
+
+function setCursor(e, save = true) {
+  let i = 0,
+    cursorlist = document.getElementById("cursor-list").firstChild;
+  while (save && cursorlist) {
+    if (cursorlist.nodeType === 1) {
+      i++;
+      cursorlist.firstChild.removeAttribute("class");
+      if (i === e) {
+        cursorlist.firstChild.classList.add("nav-selected");
+      }
+    }
+    cursorlist = cursorlist.nextSibling;
+  }
+  let cursors = [
+    "disabled",
+    "default",
+    "none",
+    "crosshair",
+    "dot",
+    "pointer",
+    "cell",
+  ];
+  if (cursors.length > e) {
+    const el = document.getElementById("remote-screen-img");
+    if (save) {
+      window.settings.cursor = e;
+    }
+    if (connectedToServer) {
```

Comment:
Does it matter if we're connected to the server? We want to avoid trying to send requests to the backend if we're disconnected, but if we're just changing around the DOM locally, it shouldn't matter.